sector-builder/src/builder.rs

```diff
@@ -68,33 +58,69 @@ impl SectorBuilder {
             let rx = Arc::new(Mutex::new(rx));
 
             let workers = (0..NUM_SEAL_WORKERS)
-                .map(|n| SealerWorker::start(n, rx.clone(), sector_store.clone(), prover_id))
+                .map(|n| SealerWorker::start(n, rx.clone(), prover_id))
                 .collect();
 
             (tx, workers)
         };
 
-        let SectorClass(sector_size, _) = sector_class;
+        let sector_size = sector_class.0.into();
 
-        // Configure main worker.
-        let main_worker = Scheduler::start_with_metadata(
-            main_rx,
-            main_tx.clone(),
-            seal_tx.clone(),
-            kv_store.clone(),
-            sector_store.clone(),
-            last_committed_sector_id,
+        // Initialize the key/value store in which we store metadata
+        // snapshots.
+        let kv_store = SledKvs::initialize(PathBuf::from(metadata_dir.clone()))
+            .expect("failed to initialize K/V store");
+
+        // Initialize a SectorStore and wrap it in an Arc so we can access it
+        // from multiple threads. Our implementation assumes that the
+        // SectorStore is safe for concurrent access.
+        let sector_store = new_sector_store(
+            sector_class,
+            sealed_sector_dir.clone(),
+            staged_sector_dir.clone(),
+        );
+
+        // Build the scheduler's initial state. If available, we
+        // reconstitute this state from persisted metadata. If not, we
+        // create it from scratch.
+        let state = {
+            let loaded =
+                crate::helpers::load_snapshot(&kv_store, &SnapshotKey::new(prover_id, sector_size))
+                    .expects(FATAL_NOLOAD)
+                    .map(Into::into);
+
+            loaded.unwrap_or_else(|| SectorBuilderState {
+                staged: StagedState {
+                    sector_id_nonce: u64::from(last_committed_sector_id),
+                    sectors: Default::default(),
+                },
+                sealed: Default::default(),
+            })
+        };
```

Comment:
would be nice to wrap this into a method, like `SectorBuilderState::new(last_committed_sector_id)`

Reply:
good call. updated.